pkg/util/v1beta1/katibconfig/config.go

```diff
@@ -34,9 +34,7 @@ import (
 
 // SuggestionConfig is the JSON suggestion structure in Katib config.
 type SuggestionConfig struct {
-	Image                     string                           `json:"image"`
-	ImagePullPolicy           corev1.PullPolicy                `json:"imagePullPolicy,omitempty"`
-	Resource                  corev1.ResourceRequirements      `json:"resources,omitempty"`
+	corev1.Container          `json:",inline"`
 	ServiceAccountName        string                           `json:"serviceAccountName,omitempty"`
 	VolumeMountPath           string                           `json:"volumeMountPath,omitempty"`
```

Comment:
@fischor Do we need `VolumeMountPath` if that is part of Container parameters: https://github.com/kubeflow/katib/blob/7f1b2842f87158c25c54f577da1b82ddc7a958e3/pkg/controller.v1beta1/suggestion/composer/composer.go#L237 ?

Reply:
The current Katib `SuggestionConfig` has a JSON field `volumeMountPath` thats just a filepath, the `corev1.Container` has a json field `volumeMounts` thats an array of volume mounts. So `volumeMountPath` is not part if the container fields.

The referenced snippet kind of follows the logic:
> ... if parameters are set by the user we should use them, otherwise use the default values.

In case the `volumeMounts` are set we prefer that, otherwise mount the default suggestion volume under the specificed `volumeMountPath`.

Reply:
I see, that is because we set [the default value name for Suggestion Volume](https://github.com/kubeflow/katib/blob/7f1b2842f87158c25c54f577da1b82ddc7a958e3/pkg/controller.v1beta1/suggestion/composer/composer.go#L121). Please can you leave comment that we should refactor `VolumeMountPath` parameter once we allow to set Deployment spec in our Suggestion config ?

Also, I think we should modify [`len(suggestionContainer.VolumeMounts) == 0` check](https://github.com/kubeflow/katib/blob/7f1b2842f87158c25c54f577da1b82ddc7a958e3/pkg/controller.v1beta1/suggestion/composer/composer.go#L237).
We always should append `VolumeMount` if ResumePolicy is `FromVolume`.

```go
if s.Spec.ResumePolicy == experimentsv1beta1.FromVolume {
volumeMount := corev1.VolumeMount{
	Name:      consts.ContainerSuggestionVolumeName,
	MountPath: suggestionConfigData.VolumeMountPath,
}
suggestionContainer.VolumeMounts = append(suggestionContainer.VolumeMounts, volumeMount)
}
```
WDYT @fischor ?

Reply:
@andreyvelich Have you seen that the consts.DefaultSuggestionPortName port is also just set if the Container specified in the suggestion has no ports set? See https://github.com/fischor/katib/blob/7f1b2842f87158c25c54f577da1b82ddc7a958e3/pkg/controller.v1beta1/suggestion/composer/composer.go#L194-L201
Would you also change that to always append the default port?

I think both ways of doing this have pros/cons. If we always append the default volume/port then the user has no option to overwrite it. If we never append the default volume/port, then the user needs to be aware that it has to specify it when setting any volumes/ports.

A third option would be to check if the default volume/port is already set in the container and only append if its not. I think I would prefer that.

Reply:
> A third option would be to check if the default volume/port is already set in the container and only append if its not. I think I would prefer that.

I think that should work, but we need to make sure that volume and port name is correct (`DefaultSuggestionPortName` and `ContainerSuggestionVolumeName`).